### PR TITLE
Initial commit

### PR DESCRIPTION
**Pull request by swatkatz**
Generate an initial commit using the `tsp init --template emitter-ts`

### CODE REVIEW

File: packages/graphql/.gitignore [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# MacOS
```
Comment (timotheeguerin) [inline packages/graphql/.gitignore:1]: coudl probably just use the top level gitignore, i think those are all already excluded

File: packages/graphql/eslint.config.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+// @ts-check
```
Comment (timotheeguerin) [inline packages/graphql/eslint.config.js:1]: we also have a standard eslint config at the top which should validate your code automatically

File: packages/graphql/package.json [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      "types": "./dist/src/index.d.ts",
+      "default": "./dist/src/index.js"
+    },
+    "./testing": {
+      "types": "./dist/src/testing/index.d.ts",
+      "default": "./dist/src/testing/index.js"
+    }
+  },
+  "dependencies": {
+    "@typespec/compiler": "^0.61.2"
```
Comment (timotheeguerin) [inline packages/graphql/package.json:17]:
```suggestion
    "@typespec/compiler": "workspace:~"
```

File: packages/graphql/package.json [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    ".": {
+      "types": "./dist/src/index.d.ts",
+      "default": "./dist/src/index.js"
+    },
+    "./testing": {
+      "types": "./dist/src/testing/index.d.ts",
+      "default": "./dist/src/testing/index.js"
+    }
+  },
+  "dependencies": {
```
Comment (timotheeguerin) [inline packages/graphql/package.json:16]:
```suggestion
  "peerDependencies": {
```

File: packages/graphql/package.json [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    "prettier": "^3.0.3"
+  },
+  "scripts": {
+    "build": "tsc",
+    "watch": "tsc --watch",
+    "test": "node --test",
+    "lint": "eslint src/ test/ --report-unused-disable-directives --max-warnings=0",
+    "lint:fix": "eslint . --report-unused-disable-directives --fix",
+    "format": "prettier .  --write",
+    "format:check": "prettier --check ."
```
Comment (timotheeguerin) [inline packages/graphql/package.json:35]: all of those are run from the workspace, you can keep them in this package if you want to be able to do a focused run

File: packages/graphql/prettierrc.yaml
Comment (timotheeguerin) [inline packages/graphql/prettierrc.yaml:1]: should use the top level one

File: packages/graphql/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,17 @@
+{
+  "compilerOptions": {
```
Comment (timotheeguerin) [inline packages/graphql/tsconfig.json:2]: would be good to extend the top level `tspconfig.base.json` and leave any overrides

File: packages/graphql/vitest.config.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+import { defineConfig } from "vitest/config";
+
+export default defineConfig({
```
Comment (timotheeguerin) [inline packages/graphql/vitest.config.ts:3]: here too you should be able to extend our top level config
```ts
import { defineConfig, mergeConfig } from "vitest/config";
import { defaultTypeSpecVitestConfig } from "../../vitest.workspace.js";

export default mergeConfig(defaultTypeSpecVitestConfig, defineConfig({}));

```

File: packages/graphql/.editorconfig [hunk 8]
```diff
@@ -0,0 +1,12 @@
+# EditorConfig is awesome: https://EditorConfig.org
```
Comment (timotheeguerin) [inline packages/graphql/.editorconfig:1]: Would be good to use top level one here

**Review by timotheeguerin — COMMENTED** [review]
Nice!

Seems like the CI is not running against feature branches, will need to fix that

File: packages/graphql/eslint.config.js (hunk 2 above)
Comment (swatkatz) [reply packages/graphql/eslint.config.js:1]: When you say automatically, do you mean I need to run `npm run lint`?

File: packages/graphql/tsconfig.json [hunk 9]
```diff
@@ -0,0 +1,17 @@
...
+    "rootDir": ".",
+    "outDir": "dist",
+    "sourceMap": true,
+    "declaration": true,
+
+    /* Linting */
+    "strict": true
+  },
+  "include": ["src", "test"]
+}
```
[the remark this answers is not in the document]
Comment (swatkatz) [reply packages/graphql/tsconfig.json:17]: Ack, will take a look

File: packages/graphql/eslint.config.js (hunk 2 above)
Comment (timotheeguerin) [reply packages/graphql/eslint.config.js:1]: yeah sorry, I meant as part of our comon CI but yeah `pnpm lint` from the top of the repo.

Most of our packages also have this
```
    "lint": "eslint . --max-warnings=0",
    "lint:fix": "eslint . --fix"
```

File: packages/graphql/package.json [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+  },
+  "peerDependencies": {
+    "@typespec/compiler": "workspace:~"
+  },
+  "devDependencies": {
+    "@types/node": "~18.11.9",
+    "source-map-support": "^0.5.21",
+    "@typescript-eslint/eslint-plugin": "^6.0.0",
+    "@typescript-eslint/parser": "^6.0.0",
+    "eslint": "^8.45.0",
```
Comment (timotheeguerin) [inline packages/graphql/package.json:24]: sorry also missed that but eslint deps and prettier can just be removed and added to the top level.

PR is also probably missing a `pnpm install` to link dependencies

File: packages/graphql/package.json (hunk 5 above)
Comment (swatkatz) [reply packages/graphql/package.json:35]: on second thoughts, will remove

**Pull request closed without merging**